draft-tjjk-cared.md

```diff
@@ -1,336 +1,405 @@
+on tunneling DNS traffic to a managed gateway or proxy whose IP
+address is known. This in turn forces enterprises to choose between
+running proxy or gateway infrastructure per client (or at least
+per-client IP address mappings) or losing client identification.
+
+Strong identification of encrypted DNS clients by the encrypted DNS
+server also brings identification up to the application layer, which
+insulates the mechanism used for identity management from network
+topology changes and allows the sense of client identity in the
+server to persist despite client IP address changes.
+
+# Drawbacks of client authentication with encrypted DNS
+
+While there are benefits in using client authentication with encrypted
+DNS in limited circumstances, authentication has drawbacks that
+make it inappropriate in many other cases. For example, client
+authentication is generally considered to be bad practice in uses
+of encrypted DNS that are motivated by client anonyminity, since
+it allows a full history of resolution requests to be associated
+confidently with the identity of an individual client. This is
+unacceptable practice.
+
+Public DNS servers generally implement allow- and blocklists at the
+IP layer as part of a layered defence against abuse; IP-layer access
+control allows unwanted traffic to be discarded early and helps
+mitigate unwanted server load. Public encrypted DNS servers with
+the additional requirement to permit authenticated clients makes
+it impossible to drop unwanted traffic early based on source IP
+address, which increases the cost of mitigation and adds complexity
+that may well introduce additional attack vectors.
+
+# When to use client authentication with encrypted DNS
+
+Encrypted DNS servers that provide resolution using transport
+protocols that incorporate TLS or dTLS such as DoH {{?RFC8484}},
+DoT {{?RFC7858}}, and DoQ {{?RFC9250}} SHOULD NOT provide client
+authentication except in the limited situations described in this
+section.
+
+Encrypted DNS servers that use other transport protocols are not
+in scope for this document.
+
+Encrypted DNS servers MUST NOT challenge clients for authentication unless they need to
+restrict connections to a set of clients they have a pre-existing relationship with as defined
+in {{#restrict-clients}}, regardless of whether or not the requirements in {{per-client}} apply.
+Encrypted DNS servers that meet the requirements in {{restrict-clients}} MAY
+challenge clients to authenticate to avoid achieving the same goal of identifying clients
+through other, less secure means (such as IP address or data in the DNS query payload).
+
+### Restricting connections to allowed clients {#restrict-clients}
+
+An encrypted DNS server that provides resolution to a specific set
+of clients and refuses service to all other clients MAY require
+clients to authenticate.
+
+For example, an encrypted DNS server owned by an enterprise that
```

Comment:
This paragraph loses a normative from the original that servers SHOULD NOT challenge clients for authentication when the server does not have a verifiable identity due to opportunistic DoT (or even RFC 9462 for IP-based discovery). We need to preserve this requirement.